examples/sampling-monomial-gamma-hmc.ipynb

```diff
@@ -0,0 +1,413 @@
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "import pints\n",
+    "import pints.toy\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "# Create log pdf\n",
+    "log_pdf = pints.toy.GaussianLogPDF([2, 4], [[1, 0], [0, 3]])\n",
+    "\n",
+    "# Contour plot of pdf\n",
+    "levels = np.linspace(-3,12,20)\n",
+    "num_points = 100\n",
+    "x = np.linspace(-1, 5, num_points)\n",
+    "y = np.linspace(-0, 8, num_points)\n",
+    "X, Y = np.meshgrid(x, y)\n",
+    "Z = np.zeros(X.shape)\n",
+    "Z = np.exp([[log_pdf([i, j]) for i in x] for j in y])\n",
+    "plt.contour(X, Y, Z)\n",
+    "plt.xlabel('x')\n",
+    "plt.ylabel('y')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we set up and run a sampling routine using Monomial-Gamma HMC MCMC"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "c:\\users\\bclamber\\desktop\\pints\\pints\\_mcmc\\_monomial_gamma_hmc.py:177: RuntimeWarning: overflow encountered in exp\n",
+      "  1.0 + np.exp(c * self._g(p, a, m)))\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Running...\n",
+      "Using Monomial-Gamma Hamiltonian Monte Carlo\n",
+      "Generating 3 chains.\n",
+      "Running in sequential mode.\n",
+      "Iter. Eval. Accept.   Accept.   Accept.   Time m:s\n",
+      "0     3      0         0         0          0:00.0\n",
+      "1     63     0.333     0.333     0.333      0:00.0\n",
+      "2     123    0.5       0.5       0.5        0:00.0\n",
+      "3     183    0.6       0.6       0.6        0:00.0\n",
+      "100   6003   0.98      0.98      0.98       0:00.6\n",
+      "200   12003  0.990099  0.990099  0.990099   0:01.1\n",
+      "300   18003  0.993     0.993     0.993      0:01.7\n",
+      "400   24003  0.995     0.995     0.995      0:02.3\n",
+      "500   30003  0.994     0.996     0.996      0:02.8\n",
+      "600   36003  0.995     0.997     0.997      0:03.4\n",
+      "700   42003  0.994302  0.997151  0.997151   0:04.0\n",
+      "800   48003  0.995     0.998     0.998      0:04.5\n",
+      "900   54003  0.996     0.998     0.998      0:05.1\n",
+      "1000  59943  0.996004  0.998002  0.998002   0:05.6\n",
+      "Halting: Maximum number of iterations (1000) reached.\n",
+      "Done!\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Choose starting points for 3 mcmc chains\n",
+    "xs = [\n",
+    "    [2, 1],\n",
+    "    [3, 3],\n",
+    "    [5, 4],\n",
+    "]\n",
+    "\n",
+    "# Set a standard deviation, to give the method a sense of scale\n",
+    "sigma = [1, 1]\n",
+    "\n",
+    "# Create mcmc routine\n",
+    "mcmc = pints.MCMCController(log_pdf, 3, xs, method=pints.MonomialGammaHMCMCMC, sigma0=sigma)\n",
```

Comment:
Can we stick the sigma arg before the method? Just to stay 'canon' in the examples :D

Reply:
Done